tests/functional/conftest.py

```diff
@@ -1,138 +1,158 @@
     # Allows access to all proxy paths - so we don't have to specify the pr proxy explicitly
-    await product.update_paths(paths=["/", "/*"])
-    return product
-
+    product.update_paths(paths=["/", "/*"], api_products=api_products)
+    LOGGER.info(f'product.get_product_details(): {product.get_product_details(api_products)}')
 
-@pytest.fixture()
-async def get_token(request):
-    """Get an access or refresh token
-    some examples:
-        1. access_token via simulated oauth (default)
-            get_token()
-        2. get access token with a specified timeout value (default is 5 seconds)
-            get_token(timeout=500000)  # 5 minuets
-        3. refresh_token via simulated oauth
-            get_token(grant_type="refresh_token", refresh_token=<refresh_token>)
-        4. access_token with JWT
-            get_token(grant_type='client_credentials', _jwt=jwt)
-        5. access_token using a specific app
-            get_token(app=<app>)
-
-    Args:
-        request(requests): HTTP requests object.
-
-    Returns:
-        _token(dict): Identity Service HTTP Response body
-        e.g. { "accessToken" : "eJkajgolJ...", "refreshToken" : "eJjagk.."}.
-    """
-    async def _token(
-        grant_type: str = "authorization_code",
-        test_app: ApigeeApiDeveloperApps = None,
-        **kwargs
-    ):
-        if test_app:
-            # Use provided test app
-            oauth = OauthHelper(test_app.client_id, test_app.client_secret, test_app.callback_url)
-            resp = await oauth.get_token_response(grant_type=grant_type, **kwargs)
-        else:
-            # Use env vars
-            oauth = OauthHelper(
-                client_id=config.CLIENT_ID,
-                client_secret=config.CLIENT_SECRET,
-                redirect_uri="https://example.org/callback",
-            )
-            resp = await oauth.get_token_response(grant_type=grant_type, **kwargs)
-
-        if resp['status_code'] != 200:
-            message = 'unable to get token'
-            raise RuntimeError(f"\n{'*' * len(message)}\n"
-                               f"MESSAGE: {message}\n"
-                               f"URL: {resp.get('url')}\n"
-                               f"STATUS CODE: {resp.get('status_code')}\n"
-                               f"RESPONSE: {resp.get('body')}\n"
-                               f"HEADERS: {resp.get('headers')}\n"
-                               f"{'*' * len(message)}\n")
-        return resp['body']
-
-    return _token
+    return product
 
 
 @pytest.fixture(scope="function")
-async def setup_session(request):
+def setup_session(request, _jwt_keys, apigee_environment, client, api_products):
     """This fixture is called at a function level.
     The default app created here should be modified by your tests.
     """
 
-    product = await _product_with_full_access()
+    product = _product_with_full_access(api_products)
+    product.update_environments([config.ENVIRONMENT], api_products=api_products)
```

Comment:
Do both `config.ENVIRONMENT` and pytest-nhds-apim's fixture `apigee_enviroment` point to the same thing? If so, could we swap out one for the other and add it to the refactoring list in [SPINEDEM-1680](https://nhsd-jira.digital.nhs.uk/browse/SPINEDEM-1680)?

Reply:
Good idea. I'll add an additional requirement to the tech-debt ticket for us to clean up the config.py and use available fixtures